core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,18 +53,48 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              condition: [/jp(e?)g$/, /\.jp(e?)g$/],
+            },
+            jpg: {
+              format: 'image/jpg',
+              condition: [/jp(e?)g$/, /\.jp(e?)g$/],
+            },
+            gif: {
+              format: 'image/gif',
+              condition: [/gif$/, /\.gif$/],
+            },
+            png: {
+              format: 'image/png',
+              condition: [/png$/, /\.png$/],
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              condition: [/svg\+xml$/, /\.svg$/],
+            }
+          };
+
+          var check1 = true;
```

Comment:
why do we need this? Also, please don't use names liek this.

Reply:
This is a simple way to check the conditions which was used earlier. I have changed the variable names.